uniffi_bindgen/src/backend/types.rs

```diff
@@ -58,6 +58,18 @@ pub trait CodeType {
         unimplemented!("Unimplemented for {}", self.type_label(oracle))
     }
 
+    /// Name of the FfiConverter
+    ///
+    /// This is the object that contains the lower, write, lift, and read methods for this type.
+    /// Depending on the binding this will either be a singleton or a class with static methods.
+    ///
+    /// This is the newer way of handling these methods and replaces the lower, write, lift, and
+    /// read CodeType methods.  Currently only used by Kotlin, but the plan is to move other
+    /// backends to using this.
+    fn ffi_converter_name(&self, oracle: &dyn CodeOracle) -> String {
+        oracle.class_name(&format!("FfiConverter{}", self.canonical_name(oracle)))
+    }
+
     /// An expression for lowering a value into something we can pass over the FFI.
     ///
     /// N.B. This should align with the `helper_code` generated by this `CodeType`.
```

Comment:
I wonder why you didn't delete L73-L105, or better/simpler: provide replace the `unimplemented!` macros with implementations.

Reply:
Coming back to this, I think default implementations should generate the expressions in terms of the `FfiConverter` methods.

If we need to customize/error out for a particular type then we still can at binding generation time, rather than wait for binding compile time or runtime.

Reply:
Agreed.  The only reason I didn't change this was that I didn't want to mess with the bindings that don't use FFIConverter yet.  Once they all get switched over, I think we can do something like this.